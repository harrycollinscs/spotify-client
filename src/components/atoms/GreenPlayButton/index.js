import React from "react";
import { faPlay } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import styled from "styled-components";

const IconContainer = styled.div`
  height: 40px;
  width: 40px;
  border-radius: 40px;
  background-color: rgb(29, 185, 84);
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
`;

const IconStyle = {
  color: "#fff",
  margin: "20px",
  height: "10px",
  width: "10px",
};

const GreenPlayButton = ({ action }) => (
  <IconContainer onClick={action}>
    <FontAwesomeIcon icon={faPlay} style={IconStyle} />
  </IconContainer>
);

export default GreenPlayButton;
