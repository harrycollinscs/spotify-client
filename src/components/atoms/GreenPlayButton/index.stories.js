import React from "react";
import GreenPlayButton from "./index";

export default {
  title: "Atoms/GreenPlayButton",
  component: GreenPlayButton,
  argTypes: {
    action: {
      action: "clicked",
    },
  },
};

const Template = (args) => <GreenPlayButton {...args} />;

export const Primary = Template.bind({});
