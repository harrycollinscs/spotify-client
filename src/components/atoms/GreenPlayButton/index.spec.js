import React from "react";
import GreenPlayButton, {IconContainer } from "./index";
import { shallow } from "enzyme";

describe("Atoms/GreenPlayButton", () => {
  const props = {
    action: jest.fn(),
  };

  const component = <GreenPlayButton {...props} />;

  it("Should match snapshot", () => {
    const render = shallow(component);
    expect(render).toMatchSnapshot();
  });

  it("Should call action on click", () => {
    const render = shallow(component);
    render.find(IconContainer).simulate("click");
    expect(props.action).toHaveBeenCalled();
  });
});
