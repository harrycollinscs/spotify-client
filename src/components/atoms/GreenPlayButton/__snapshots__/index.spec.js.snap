// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/GreenPlayButton Should match snapshot 1`] = `
<styled.div
  onClick={[MockFunction]}
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "icon": Array [
          448,
          512,
          Array [],
          "f04b",
          "M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z",
        ],
        "iconName": "play",
        "prefix": "fas",
      }
    }
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    style={
      Object {
        "color": "#fff",
        "height": "10px",
        "margin": "20px",
        "width": "10px",
      }
    }
    swapOpacity={false}
    symbol={false}
    title=""
    transform={null}
  />
</styled.div>
`;
